lib/polymorphic_integer_type/activerecord_5_0_0/polymorphic_array_value_extension.rb

```diff
@@ -14,7 +14,7 @@ def type_to_ids_mapping
       name = association.name
       default_hash = Hash.new { |hsh, key| hsh[key] = [] }
       values.each_with_object(default_hash) do |value, hash|
-        klass = respond_to?(:klass) ? klass(value) : klass.class
+        klass = respond_to?(:klass, true) ? klass(value) : value.class
```

Comment:
this was just straight up broken before. We need 
- `respond_to?(:klass, true)` because `#klass` is private
- `value.class` instead of `klass.class` since klass is undefined

Reply:
The reason why it doesn't care about using `send` to call it is that you're calling it on `self`. Method visibility only kicks in when you're calling a private method on a _different_ object.